lib/streamlit/watcher/LocalSourcesWatcher.py

```diff
@@ -158,16 +159,13 @@ def update_watched_modules(self):
                     # .origin is 'built-in'.
                     continue
 
-                is_in_blacklisted_folder = any(
-                    _file_is_in_folder(filepath, blacklisted_folder)
-                    for blacklisted_folder in self._folder_blacklist
-                )
-
-                if is_in_blacklisted_folder:
+                if self._folder_black_list.is_blacklisted(filepath):
                     continue
 
                 file_is_new = filepath not in self._watched_modules
-                file_is_local = _file_is_in_folder(filepath, self._report.script_folder)
+                file_is_local = util.file_is_in_folder(
+                    filepath, self._report.script_folder
+                )
```

Comment:
Same as `hashing.py`, I am leaving i) checking if the file is local and ii) checking against the blacklist separate.
Also, here we use then ` self._report.script_folder` as the main script folder.